jdaviz/configs/default/plugins/export_plot/export_plot.py

```diff
@@ -1,23 +1,60 @@
 from jdaviz.core.registries import tray_registry
 from jdaviz.core.template_mixin import PluginTemplateMixin, ViewerSelectMixin
+from jdaviz.core.user_api import PluginUserApi
 
 __all__ = ['ExportViewer']
 
 
 @tray_registry('g-export-plot', label="Export Plot")
 class ExportViewer(PluginTemplateMixin, ViewerSelectMixin):
+    """
+    See the :ref:`Export Plot Plugin Documentation <imviz-export-plot>` for more details.
+
+    Only the following attributes and methods are available through the
+    :ref:`public plugin API <plugin-apis>`:
+
+    * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.show`
+    * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.open_in_tray`
+    * :meth:`save_figure`
+    """
     template_file = __file__, "export_plot.vue"
 
+    @property
+    def user_api(self):
+        return PluginUserApi(self, expose=('save_figure',))
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
-    def vue_save_figure(self, filetype):
+    def save_figure(self, filename=None, filetype=None):
         """
-        Callback for save figure events in the front end viewer toolbars. Uses
-        the bqplot.Figure save methods.
+        Save the figure to an image with a provided filename or through an interactive save dialog.
+
+        Parameters
+        ----------
+        filename : str or None
+            Filename to autopopulate the save dialog.
+        filetype : str or None
+            Filetype (png or svg).  If None, will default based on ``filename`` or to 'png'.
         """
+        if filetype is None:
+            if filename is not None:
+                filetype = filename.split('.')[-1]

```

Comment:
Sometimes people just give the name without suffix because they are used to other software attaching suffix for them.

Reply:
Maybe the code to auto populate the filename should be its own private function, so at least we can test that part. The only thing we cannot test is `viewer.figure.save_png()` call.

Reply:
that's fine, this logic should handle that scenario as expected, right?  I guess the only problem would be if they did `filename='blah.randomextension'` which would raise a `NotImplementedError` instead of saving to `blah.randomextension.png`.  

Should I instead only apply the filetype if its either png or svg and otherwise default to png?  But then `filename='blah.jpeg'` would silently write to `blah.jpeg.png`.

Reply:
Silent fallback means they still get the image, better than error from trying to save to an unsupported extension? Unless you want upstream to deal with that instead?

Reply:
I always prefer (helpful) errors than unexpected magic/fallbacks myself 🤷

Reply:
Okay, I'll leave it up to your good judgement then.

Reply:
I think we should have a separate error here for when the length of the split string is 1. The problem in that case isn't that I've specified an invalid filetype, it's that I haven't specified one at all.

Reply:
@rosteen - with the current logic, that case should still just silently default to png (and this is stated in the API docstring).  You know I always prefer raising errors over making assumptions... so I'm very tempted by your suggestion 😉 but think here it might just get in the way and its common for save dialogs to have a default extension.  The question is, what should happen when calling `plugin.save_figure('myimage')` - an error or saving to `myimage.png`?

Reply:
I made that comment because I just tried it and got the `NotImplementedError`:

```
ep.save_figure(filename="test_spec")

executed in 364ms, finished 10:37:50 2022-09-30
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
Cell In [11], line 1
----> 1 ep.save_figure(filename="test_spec")

File ~/projects/jdaviz/jdaviz/configs/default/plugins/export_plot/export_plot.py:54, in ExportViewer.save_figure(self, filename, filetype)
     52     viewer.figure.save_svg(filename)
     53 else:
---> 54     raise NotImplementedError(f"filetype={filetype} not supported")

NotImplementedError: filetype=test_spec not supported


```

Reply:
ah, should have tested myself!  So should that raise an error or default to png? 😂

Reply:
I vote to raise the error, mostly for future-proofing in case we're later able to implement additional filetypes for exporting in this plugin that are more different than `PNG` vs `SVG`.

Reply:
It's pretty low stakes though, and it sounds like @pllim would prefer the automatic fallback.

Reply:
When I don't give file extension to Photoshop, it doesn't crash. It saves my stuff in some default format. I think users might be looking for similar establish behaviors.

Reply:
Works for me

Reply:
done (default to png when no extension is provided in the filename by checking for `'.' in filename`)

Reply:
By the way, would mimetypes be useful for your purpose? https://docs.python.org/3/library/mimetypes.html

```python
In [10]: mimetypes.guess_type('dummy_gwcs.png')
Out[10]: ('image/png', None)

In [11]: mimetypes.guess_type("test_spec")
Out[11]: (None, None)
```

Reply:
interesting idea.  I think it might be overkill here since we're passing to one of two backend calls... but maybe if we generalize that in the future to avoid the file dialog, this would be a good direction to take.